# N_terminus_MSA_check

Using modern genome assembly tools, scientists can annotate genomes in record time. This revolution in genome assembly is not without its own problems however. Weak signals along with incomplete sequencing reads can create a problem of gene combination. This is an error in identification of genes that includes a lot of noise from another gene by accidental combination. Using genes of similar sequence and alignment tools we can catch these outliers and hopefully remove these imperfections from genomes.

My program seeks to solve this problem by using multiple sequence alignment data from hundreds of genomes to identify misidentified gene start sites. 

Orthologous groups and Multiple Sequence Alignments are pulled from:
  https://omabrowser.org/
  http://eggnog5.embl.de/
